build-support/bin/release.sh

```diff
@@ -588,20 +593,22 @@ function build_pex() {
           exit 1
           ;;
       esac
-      local dest="${ROOT}/dist/pants.${PANTS_UNSTABLE_VERSION}.${platform}.pex"
-      local stable_dest="${DEPLOY_DIR}/pex/pants.${PANTS_STABLE_VERSION}.${platform}.pex"
+      local dest="${ROOT}/dist/pants.${PANTS_UNSTABLE_VERSION}.${platform}.${dest_suffix}"
+      local stable_dest="${DEPLOY_DIR}/pex/pants.${PANTS_STABLE_VERSION}.${platform}.${dest_suffix}"
       ;;
     fetch)
-      # NB: We do not include Python 3 wheels, as we cannot release a Python 3 compatible PEX
-      # until https://github.com/pantsbuild/pex/issues/654 is fixed.
       local platforms=()
+      abis=("cp-27-mu" "cp-27-m")
+      if [[ "${python_three:-false}" == "true" ]]; then
+        abis=("cp-36-abi3")
```

Comment:
@jsirois I cannot figure out what to put here.

If we set `("cp-36-m")`, then cryptography will fail to resolve because it is `cp34-abi3`. If we put `("cp-36-abi3")`, all of our `cp36-cp36m` wheels like `setproctitle` and `psutil` fail to resolve.

Putting both `("cp-36-m" "cp-36-abi3")` fails, as multiple `--platform` argues seems to result in saying for each platform, get every wheel needed, rather than for each platform, get as many wheels as you can.

Is the solution to set to `"cp3-36-m"` and to teach upstream PEX to accept `abi3` for any version `<=` the specified version? For example, `cp36-cp36m` should accept `cp34-abi3` and `cp36-abi3` but not `cp37-abi3`.

Reply:
The short term solution is to ensure the prerequisites to _build_ cryptography. PEX is configured to try, in order, wheel, egg, sdist. Cryptography has an sdist: https://pypi.org/project/cryptography/#files

Reply:
> The short term solution is to ensure the prerequisites to build cryptography. PEX is configured to try, in order, wheel, egg, sdist. Cryptography has an sdist: https://pypi.org/project/cryptography/#files

To clarify, we would need to build this wheel on the wheel builder shards, right? We couldn't build it within the `release -p` call in these deploy shards, because they use Ubuntu so don't have max compatibility and we would be missing the OSX shards.

If so, that seems difficult to me to figure out how to get the wheel builder shards to not use the totally valid `cp-34-abi3` wheel that exists. I have no idea how we would do that.

--

Instead, would the original proposal work?

> Is the solution to set to "cp3-36-m" and to teach upstream PEX to accept abi3 for any version <= the specified version? For example, cp36-cp36m should accept cp34-abi3 and cp36-abi3 but not cp37-abi3.

Reply:
Hm, edit. Looks like there's a command in pip to build and not use bdist: `https://cryptography.io/en/latest/installation/#building`. Can experiment with this if you think it will be easier than upstream change proposed.